src/Microsoft.Health.Fhir.TemplateManagement/ArtifactProviders/BlobTemplateProvider.cs

```diff
@@ -0,0 +1,99 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using DotLiquid;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Health.Fhir.Liquid.Converter.Utilities;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.ArtifactProviders
+{
+    public class BlobTemplateProvider : IConvertDataTemplateProvider
+    {
+        private BlobContainerClient _blobContainerClient;
+
+        private readonly IMemoryCache _templateCache;
+
+        private readonly string _blobTemplateCacheKey = "cached-blob-templates";
+
+        private readonly int _segmentSize = 100;
+
+        public BlobTemplateProvider(BlobContainerClient blobContainerClient, IMemoryCache templateCache)
```

Comment:
It looks like BlobContainerClient should be mockable, https://devblogs.microsoft.com/azure-sdk/unit-testing-and-mocking/

If so we should have unit tests for this class to verify the logic.

Reply:
Thanks, I've added unit tests